livekit-agents/livekit/agents/tts/fallback_adapter.py

```diff
@@ -0,0 +1,153 @@
+            )
+        raise NotImplementedError(
+            "streaming is not supported by this TTS, please use a different TTS or use a StreamAdapter"
+        )
+
+
+class FallbackChunkedStream(ChunkedStream):
+    def __init__(
+        self,
+        *,
+        providers: tuple[TTS, ...],
+        connect_timeout: float,
+        keepalive_timeout: float,
+        text: str,
+    ) -> None:
+        super().__init__()
+        self._text = text
+        self._connect_timeout = connect_timeout
+        self._keepalive_timeout = keepalive_timeout
+        self._providers = providers
+
+    @utils.log_exceptions(logger=logger)
+    async def _main_task(self) -> None:
+        for provider in self._providers:
+            stream = provider.synthesize(self._text)
+            timeout = self._connect_timeout
+            while True:
+                try:
+                    item = await asyncio.wait_for(stream.__anext__(), timeout)
+                    timeout = self._keepalive_timeout
+                    self._event_ch.send_nowait(item)
+                except Exception as e:
+                    if isinstance(e, asyncio.TimeoutError):
+                        break
+                    return
+            logger.warn(
+                f"provider {provider.__class__.__module__} failed, attempting to switch"
+            )
+        raise Exception("all providers failed")
+
+
+class FallbackSynthesizeStream(SynthesizeStream):
+    def __init__(
+        self,
+        *,
+        providers: tuple[TTS, ...],
+        connect_timeout: float,
+        keepalive_timeout: float,
+    ) -> None:
+        super().__init__()
+        self._connect_timeout = connect_timeout
+        self._keepalive_timeout = keepalive_timeout
+        self._providers = providers
+        self._timeout: float
+        self._current_provider = 0
+        self._init_tts()
+
+    def _init_tts(self) -> None:
+        self._stream = self._providers[self._current_provider].stream()
+        self._timeout = self._connect_timeout
+
+    @utils.log_exceptions(logger=logger)
+    async def _main_task(self) -> None:
+        async def _pass_input():
+            async for input in self._input_ch:
+                self._stream.push_text(input)
+
+        async def _pass_output():
+            while True:
+                try:
+                    # TODO(nbsp): elevenlabs _stream._run_ws()#_recv_task() simply exits when it's
+                    #             done, and the iterator hangs indefinitely until it times out.
+                    #             probably needs a change to SynthesizeStream to make it work
+                    item = await asyncio.wait_for(
+                        self._stream.__anext__(), self._timeout
+                    )
+                    self._timeout = self._keepalive_timeout
+                    self._event_ch.send_nowait(item)
+                except Exception as e:
+                    if isinstance(e, asyncio.TimeoutError) or isinstance(
```

Comment:
This exception is never catched. asyncio.TimeoutError doesn't inherit from Exception